pilot/pkg/xds/sds.go

```diff
@@ -304,8 +309,58 @@ func toEnvoyCaSecret(name string, cert []byte) *discovery.Resource {
 	}
 }
 
-func toEnvoyKeyCertSecret(name string, key, cert []byte) *discovery.Resource {
-	res := util.MessageToAny(&envoytls.Secret{
+func toEnvoyKeyCertSecretWithPrivateKeyProvider(name string, key, cert []byte, pkpConf *mesh.PrivateKeyProvider) *anypb.Any {
```

Comment:
this has quite some duplicate code as the other `toEnvoyKeyCertSecretWithPrivateKeyProvider()`, could we refactor and try to share the common logic

Reply:
It has some, but not completely. Anyway I am working on QAT private key provider extension support. I will fine tune in that PR.